types/stylelint/index.d.ts

```diff
@@ -814,6 +814,38 @@ declare namespace stylelint {
 		column: number;
 	};
 
+	export type ProblemLocation =
+		| {
+				/**
+				 * The inclusive start position of the problem, relative to the
+				 * node's source text. If provided, this will be used instead of
+				 * `index`.
+				 */
+				start: Position;
+				/**
+				 * The exclusive end position of the problem, relative to the
+				 * node's source text. If provided, this will be used instead of
+				 * `endIndex`.
+				 */
+				end: Position;
+		  }
```

Comment:
I suppose you didn't reuse `Range` at l251 so that you could add those comments?

Reply:
I kept the original code as it already exists, only moved it around.
See: https://github.com/stylelint/stylelint/blob/727bb1661d148cb20dc9161883d945b403e70998/types/stylelint/index.d.ts#L836-L847